src/main/kotlin/biokotlin/featureTree/TypeSchema.kt

```diff
@@ -0,0 +1,390 @@
+package biokotlin.featureTree
+
+import java.util.*
+
+// This file is incomplete and will require a near total rework due to some complexities in the underlying SO file.
+
+const val SO_PATH = "src/main/resources/featureTree/so.obo"
+private interface Type {
+    val names: Set<String>
+    val isA: Set<Type>
+    val partOf: Set<Type>
+}
+
+private class BaseSchema private constructor() {
+    val baseRegistry = mutableMapOf<String, MutableSet<BaseType>>()
+    inner class BaseType(
+        override val names: Set<String>,
+        override val isA: MutableSet<BaseType>,
+        override val partOf: MutableSet<BaseType>
+    ): Type {
+        /**
+         * INVARIANTS:
+         * 1. All names point to this in [baseRegistry]
+         */
+        fun invariants(): Boolean {
+//            names.forEach {
```

Comment:
You may want to clean up these commented out blocks of code.  If you are trying to keep these for later implementation, you may want to add TODOs  or appropriate exceptions if these functions are accidentally called.